pkg/whitesource/whitesource.go

```diff
@@ -144,37 +142,25 @@ func (s *System) GetProjectsMetaInfo(productToken string) ([]Project, error) {
 		ProductToken: productToken,
 	}
 
-	respBody, err := s.sendRequest(req)
+	err := s.sendRequestAndDecodeJSON(req, &wsResponse)
 	if err != nil {
 		return nil, errors.Wrap(err, "WhiteSource request failed")
 	}
 
-	err = json.Unmarshal(respBody, &wsResponse)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to parse WhiteSource response")
-	}
-
 	return wsResponse.ProjectVitals, nil
 }
 
 // GetProjectToken returns the project token for a project with a given name
 func (s *System) GetProjectToken(productToken, projectName string) (string, error) {
-	var token string
 	project, err := s.GetProjectByName(productToken, projectName)
 	if err != nil {
 		return "", err
 	}
-
-	// returns a nil token and no error if not found
-	if project != nil {
-		token = project.Token
-	}
-
-	return token, nil
+	return project.Token, nil
 }
 
-// GetProjectVitals returns project meta info given a project token
-func (s *System) GetProjectVitals(projectToken string) (*Project, error) {
+// GetProjectByToken returns project meta info given a project token
+func (s *System) GetProjectByToken(projectToken string) (Project, error) {
```

Comment:
why not keeping the name as before? Also return value changed.
-> incompatible change breaking the current usage of this API

Reply:
Again, there were two methods doing the exact same thing. I kept only one. I did look at outside places where this is used, and not all of the duplicated methods are used even. It should be a one time effort to adapt the callers. I think it is much better than offering a confusing API with duplicated methods that sometimes return a value, sometimes nil for seemingly the same thing.

Reply:
ah ok, did not see that there was a duplicate ...